comid/integrityregisters_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package comid
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/veraison/swid"
+)
+
+func prepareRegister(t *testing.T, iType string) (*IntegrityRegisters, error) {
+	var err error
+	val := MustHexDecode(t, "e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75")
+	entry := swid.HashEntry{HashAlgID: swid.Sha256, HashValue: val}
+	reg := NewIntegrityRegisters()
+	for index := 0; index < 5; index++ {
+		switch iType {
+		case "uint":
+			err = reg.AddDigest(uint(index), entry)
+			require.NoError(t, err)
+		case "text":
+			i := fmt.Sprint(index)
+			err = reg.AddDigest(i, entry)
+			require.NoError(t, err)
+		default:
+			err = fmt.Errorf("invalid iType = %s", iType)
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+	return reg, nil
+}
+
+func TestIntegrityRegisters_AddDigest_OK(t *testing.T) {
+	_, err := prepareRegister(t, "uint")
+	require.NoError(t, err)
+}
+
+func TestIntegrityRegisters_AddDigest_NOK(t *testing.T) {
+	expectedErr := `no register to add digest`
+	register := IntegrityRegisters{}
+	err := register.AddDigest(uint(0), swid.HashEntry{})
+	assert.EqualError(t, err, expectedErr)
+	expectedErr = `unexpected type for index: bool`
+	var k bool
+	reg, err := prepareRegister(t, "uint")
+	require.NoError(t, err)
+	err = reg.AddDigest(k, swid.HashEntry{})
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestIntegrityRegisters_AddDigests_NOK(t *testing.T) {
+	expectedErr := `no digests to add`
+	register := IntegrityRegisters{}
+	err := register.AddDigests(uint(0), []swid.HashEntry{})
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestIntegrityRegisters_MarshalCBOR_UIntIndex_OK(t *testing.T) {
+	reg, err := prepareRegister(t, "uint")
+	// Below is the partial CBOR Pretty notation to highlight index as uint: unsigned(0) (at line 70)
```

Comment:
just noting that the "at line 70" comment is going to become stale quite quickly ;-)